src/app/public/modules/phone-field/phone-field-input.directive.ts

```diff
@@ -0,0 +1,237 @@
+import {
+  NG_VALUE_ACCESSOR,
+  NG_VALIDATORS,
+  ControlValueAccessor,
+  Validator,
+  AbstractControl,
+  NgControl,
+  FormControl
+} from '@angular/forms';
+
+import {
+  SkyCountryData
+} from './types';
+
+import {
+  SkyPhoneFieldComponent
+} from './phone-field.component';
+
+// tslint:disable:no-forward-ref no-use-before-declare
+const SKY_PHONE_FIELD_VALUE_ACCESSOR = {
+  provide: NG_VALUE_ACCESSOR,
+  useExisting: forwardRef(() => SkyPhoneFieldInputDirective),
+  multi: true
+};
+
+const SKY_PHONE_FIELD_VALIDATOR = {
+  provide: NG_VALIDATORS,
+  useExisting: forwardRef(() => SkyPhoneFieldInputDirective),
+  multi: true
+};
+// tslint:enable
+
+@Directive({
+  selector: '[skyPhoneFieldInput]',
+  providers: [
+    SKY_PHONE_FIELD_VALUE_ACCESSOR,
+    SKY_PHONE_FIELD_VALIDATOR
+  ]
+})
+export class SkyPhoneFieldInputDirective implements OnInit, AfterViewInit,
+  ControlValueAccessor, Validator {
+
+  /**
+   * Disabling linting here as per the Angular style guide "Style 05-13" it is acceptable
+   * to alias when when the directive name is also an input property,
+   * and the directive name doesn't describe the property.
+   */
+  // tslint:disable-next-line:no-input-rename
+  @Input('skyPhoneFieldInput')
+  public skyPhoneFieldComponent: SkyPhoneFieldComponent;
+
+  @Input()
+  public set defaultCountry(value: string) {
+    this.skyPhoneFieldComponent.setDefaultCountry(value);
+    this._defaultCountryCode = value;
+  }
+
+  public get defaultCountry() {
+    return this._defaultCountryCode;
+  }
+
+  @Input()
+  public set disabled(value: boolean) {
+    this.skyPhoneFieldComponent.disabled = value;
+    this.renderer.setProperty(
+      this.elRef.nativeElement,
+      'disabled',
+      value);
+    this._disabled = value;
+  }
+
+  public get disabled(): boolean {
+    return this._disabled || false;
+  }
+
+  @Input()
+  public formatModel: boolean = true;
+
+  @Input()
+  public noValidate: boolean = false;
+
+  @Output()
+  public selectedCountryChanged = new EventEmitter<SkyCountryData>();
+
+  private set modelValue(value: string) {
+    this._modelValue = value;
+    this.setInputValue(value);
+    if (value) {
+      let formattedValue = value;
+      if (this.formatModel) {
+        formattedValue = this.skyPhoneFieldComponent.formatNumber(value.toString());
+      }
+      this._onChange('+' + this.skyPhoneFieldComponent.selectedCountry.dialCode + ' ' + formattedValue);
+    } else {
+      this._onChange(value);
+    }
+    this._validatorChange();
+  }
+
+  private get modelValue(): string {
+    return this._modelValue;
+  }
+
+  private _defaultCountryCode: string;
+
+  private _disabled: boolean;
+
+  private _modelValue: string;
+
+  public constructor(
+    private changeDetector: ChangeDetectorRef,
+    private elRef: ElementRef,
+    private injector: Injector,
+    private renderer: Renderer2
+  ) { }
+
+  public ngOnInit(): void {
+    this.renderer.addClass(this.elRef.nativeElement, 'sky-form-control');
```

Comment:
I notice we add classes and attributes a couple times in this directive. Should we be using an adapter class to handle this?

Reply:
That's actually a good point. The date and time pickers don't use an adapter but I feel like you are probably right. @Blackbaud-SteveBrush do you have any thoughts here? I think Alex is correct but am curious if you know what the date and time pickers don't use an adapter.

Reply:
Any time a component references the DOM, we're technically supposed to create a DOM adapter service. The inconsistencies we see in the other components (not only date/time picker) are either a result of laziness, or the fact that they were written before the style guide was agreed to.

Reply:
Sounds great. Just added an adapter.